server/src/packages/rwcds-form/rwcds-form.service.ts

```diff
@@ -166,52 +171,74 @@ export class RwcdsFormService {
 
-      // IF the zr type does not exist yet, then POST the action
-      // NOTE: Some actions that exist in dcp_projectaction SHOULD NOT be copied over.
-      // The ZONING_RESOLUTION_TYPES lookup also functions as a list of actions that SHOULD be
-      // copied over from dcp_projectaction, which is why we check that `code` exists here.
-      if (!matchingZr && code) {
-        return this.crmService.create(`dcp_affectedzoningresolutions`, {
-          'dcp_zoningresolutiontype': code, // coded value
-          'dcp_zrsectionnumber': currentActionZoningResolution,
-          'dcp_modifiedzrsectionnumber': action.dcp_zrmodifyingzrtxt,
-          'dcp_rwcdsform@odata.bind': `/dcp_pasforms(${dcp_rwcdsformid})`,
-        });
-      }
-    }));
+        // action attribute that matches dcp_zrsectionnumber on dcp_affectedzoningresolution
+        const currentActionZoningResolution = action.dcp_ZoningResolution ? action.dcp_ZoningResolution.dcp_zoningresolution : null
+
+        if (!action.dcp_ZoningResolution) console.log(`No currentActionZoningResolution for Project Action Label ${projectActionLabel}, no action.dcp_ZoningResolution found.`);
+
+        if (!label) console.log(`Could not find Affected ZR Type for ${projectActionLabel}`);
+
+        // IF the action.ZoningResolution.dcp_zoningresolution value has changed since the initial POST,
+        // then PATCH the new value to dcp_zrsectionnumber on dcp_affectedzoningresolution entity
+        // NOTE: dcp_modifiedzrsectionnumber not included here because it's updated by applicant on rwcds-form
+        if (matchingZr && matchingZr.dcp_zrsectionnumber !== currentActionZoningResolution) {
+          try {
+            return this.crmService.update('dcp_affectedzoningresolutions', matchingZr.dcp_affectedzoningresolutionid, {
+              dcp_zrsectionnumber: currentActionZoningResolution,
+            });
+          } catch (e) {
+            console.log('cannot update dcp_affectedzoningresolutions for matchingZr && matchingZr.dcp_zrsectionnumber', e);
+            throw e;
+          }
+        }
+
+        // IF the zr type does not exist yet, then POST the action
+        // NOTE: Some actions that exist in dcp_projectaction SHOULD NOT be copied over.
+        // The ZONING_RESOLUTION_TYPES lookup also functions as a list of actions that SHOULD be
+        // copied over from dcp_projectaction, which is why we check that `code` exists here.
+        if (!matchingZr && code) {
+          try {
+            return this.crmService.create(`dcp_affectedzoningresolutions`, {
+              'dcp_zoningresolutiontype': code, // coded value
+              'dcp_zrsectionnumber': currentActionZoningResolution,
+              'dcp_modifiedzrsectionnumber': action.dcp_zrmodifyingzrtxt, // text was too long, need to trim to 25 char or less. need to limit text limit in crm
```

Comment:
no need to include this comment. Let's see how Aline and Liz think we should handle this, possibly on the CRM side

Reply:
I spoke to them in chat and on ticket.  Liz is looking into how this happened as there is a text limit in that input field. Aline believed that is a DCP entered field. This moves the focus to the CRM side of things.